### PR TITLE
Improve test hook, make test hook and suite container friendly (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
cherry-picked/adapted from work for #51

### CODE REVIEW

File: irods_consortium_continuous_integration_test_hook.py [hunk 1]
```diff
@@ -341,25 +321,36 @@ def main():
...
+        )
+
+        add_alias_to_etc_hosts()
+
+        install_testing_dependencies()
+
+        configure_kerberos_for_testing()
 
-    time.sleep(10)
+    test = options.test or 'test_irods_auth_plugin_krb'
```
Comment (korydraughn) [inline irods_consortium_continuous_integration_test_hook.py:345]: Consider renaming `test` (how about `test_name`)?

Feels a little loose.

File: irods_consortium_continuous_integration_test_hook.py [hunk 2]
```diff
@@ -19,17 +19,17 @@ def kdc_database_master_key():
...
         with open('/etc/hosts', 'r') as hosts_file:
             for l in hosts_file:
-                if fullname in l:
-                    hosts_copy.write(l.strip() + ' ' + shortname + '\n')
+                # add the alias to the localhost line so that the server self-identifies as the KDC host
+                if '127.0.0.1' in l and alias not in l:
+                    hosts_copy.write('{} {}\n'.format(l.strip(), alias).encode())
                 else:
-                    hosts_copy.write(l)
+                    hosts_copy.write(l.encode())
```
Comment (korydraughn) [inline irods_consortium_continuous_integration_test_hook.py:32]: Just wondering... Should `encode()` take the encoding as a string or does it default to `utf-8` or some other value?

**Review by korydraughn — COMMENTED** [review]
Looks good.

File: irods_consortium_continuous_integration_test_hook.py (hunk 2 above)
Comment (alanking) [reply irods_consortium_continuous_integration_test_hook.py:32]: https://docs.python.org/2/library/stdtypes.html?highlight=encode#str.encode

In python2, `encoding` is an optional parameter which defaults to "the current default string encoding." In python3, `encoding` defaults to "utf-8" explicitly, so I guess we will need to indicate this for the 4-2-stable version.

File: irods_consortium_continuous_integration_test_hook.py (hunk 1 above)
Comment (alanking) [reply irods_consortium_continuous_integration_test_hook.py:345]: I've created a separate issue for this in the testing environment: https://github.com/irods/irods_testing_environment/issues/81

File: irods_consortium_continuous_integration_test_hook.py (hunk 2 above)
Comment (alanking) [reply irods_consortium_continuous_integration_test_hook.py]: Done

**Review by korydraughn — APPROVED** [review]
Looks good to me.

If you're happy with it and the tests pass, merge it.

**Pull request merged**